src/notation/internal/notationplayback.h

```diff
@@ -47,15 +46,21 @@ class NotationPlayback : public INotationPlayback, public async::Asyncable
     int secToTick(float sec) const override;
 
     QRect playbackCursorRectByTick(int tick) const override;
+    QRect loopCursorRectByTick(LoopCursorType cursorType, int tick) const override;
 
     RetVal<int> playPositionTick() const override;
     void setPlayPositionTick(int tick) override;
-    bool setPlayPositionByElement(const Element* e) override;
+    bool setPlayPositionByElement(const Element* element) override;
     async::Channel<int> playPositionTickChanged() const override;
 
-    midi::MidiData playElementMidiData(const Element* e) const override;
+    midi::MidiData playElementMidiData(const Element* element) const override;
+
+    void addLoopInToSelectedNote() override;
+    void addLoopOutToSelectedNote() override;
```

Comment:
we can combine addLoopInToSelectedNote and addLoopOutToSelectedNote to one with input parameter loopType

Reply:
fixed